src/eos-shard-writer-v2.c

```diff
@@ -328,7 +331,7 @@ write_blob (EosShardWriterV2 *self, struct eos_shard_writer_v2_blob_entry *blob,
     offset += size;
   }
 
-  close (blob_fd);
+  g_assert (close (blob_fd) == 0 || errno == EINTR);
```

Comment:
I think this pattern is used pretty consistently throughout eos-shard, I'll open a ticket to address it